### PR TITLE
[script.module.autocompletion] 2.0.0

### PR DESCRIPTION
**Pull request by sualfred**
### Description

Python 3 version of Phil's autocompletion module.
Dropped not working Netflix API.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by sualfred** [comment]
@phil65 fyi

File: script.module.autocompletion/lib/AutoCompletion.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+
+def get_http(url=None, headers=False):
+    """
+    fetches data from *url, returns it as a string
+    """
+    succeed = 0
+    if not headers:
+        headers = {'User-agent': 'XBMC/16.0 ( phil65@kodi.tv )'}
+    while (succeed < 2) and (not xbmc.abortRequested):
```
Comment (enen92) [inline script.module.autocompletion/lib/AutoCompletion.py]: monitor.abortRequested()

File: script.module.autocompletion/lib/AutoCompletion.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        results = read_from_file(path)
+        log("loaded file for %s. time: %f" % (url, time.time() - now))
+    else:
+        response = get_http(url, headers)
+        try:
+            results = json.loads(response)
+            log("download %s. time: %f" % (url, time.time() - now))
+            save_to_file(results, hashed_url, cache_path)
+        except Exception as error:
+            log(error)
```
Comment (enen92) [inline script.module.autocompletion/lib/AutoCompletion.py]: Really needed all this info?

File: script.module.autocompletion/lib/AutoCompletion.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        return []
+
+
+def get_http(url=None, headers=False):
+    """
+    fetches data from *url, returns it as a string
+    """
+    succeed = 0
+    if not headers:
+        headers = {'User-agent': 'XBMC/16.0 ( phil65@kodi.tv )'}
```
Comment (enen92) [inline script.module.autocompletion/lib/AutoCompletion.py]: Why this headers are any different from the global on the top of the file? Also xbmc 16.0

File: script.module.autocompletion/lib/AutoCompletion.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+def save_to_file(content, filename, path=""):
+    """
+    dump json and save to *filename in *path
+    """
+    if not xbmcvfs.exists(path):
+        xbmcvfs.mkdirs(path)
+    text_file_path = os.path.join(path, filename + ".txt")
+    now = time.time()
+    text_file = xbmcvfs.File(text_file_path, "w")
```
Comment (enen92) [inline script.module.autocompletion/lib/AutoCompletion.py]: Use with statement here too

File: script.module.autocompletion/addon.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.module.autocompletion" name="AutoCompletion library" version="2.0.0" provider-name="Philipp Temminghoff (phil65),sualfred">
+    <requires>
+        <import addon="xbmc.python" version="3.0.0" />
+        <import addon="script.module.requests" version="2.9.1"/>
+    </requires>
+    <extension point="xbmc.python.module" library="lib" />
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en">Module providing some AutoCompletion functions</summary>
```
Comment (enen92) [inline script.module.autocompletion/addon.xml]: en_gb

File: script.module.autocompletion/addon.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        <description lang="en">Module providing some AutoCompletion functions</description>
```
Comment (enen92) [inline script.module.autocompletion/addon.xml]: en_GB

File: script.module.autocompletion/changelog.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
+1.1.2
```
Comment (enen92) [inline script.module.autocompletion/changelog.txt]: Not needed in the submission

File: script.module.autocompletion/lib/AutoCompletion.py (hunk 1 above)
Comment (sualfred) [reply script.module.autocompletion/lib/AutoCompletion.py]: done

File: script.module.autocompletion/lib/AutoCompletion.py (hunk 2 above)
Comment (sualfred) [reply script.module.autocompletion/lib/AutoCompletion.py]: nope, removed

File: script.module.autocompletion/lib/AutoCompletion.py (hunk 3 above)
Comment (sualfred) [reply script.module.autocompletion/lib/AutoCompletion.py]: was already there. changed.

File: script.module.autocompletion/lib/AutoCompletion.py (hunk 4 above)
Comment (sualfred) [reply script.module.autocompletion/lib/AutoCompletion.py]: done

File: script.module.autocompletion/addon.xml (hunk 5 above)
Comment (sualfred) [reply script.module.autocompletion/addon.xml]: done

File: script.module.autocompletion/addon.xml (hunk 6 above)
Comment (sualfred) [reply script.module.autocompletion/addon.xml]: done

File: script.module.autocompletion/changelog.txt (hunk 7 above)
Comment (sualfred) [reply script.module.autocompletion/changelog.txt]: file removed

**Pull request merged**